### PR TITLE
chore(kafka sink)!: Use comma delim server list instead of an array

### LINKED ISSUE #1004: Kafka broker configuration is not consistent across the kafka source and sink
This is feedback from a production user:

> It would be nice if the brokers were specified in the same way with the kafka sink and the kafka source.. Right now, one of them is a comma separated string and the other is actually broken out in square brackets.

Basically, the `kafka` source takes a comma-separated list of `bootstrap_servers` and the `kafka` sink takes a proper array. I prefer the array unless there is a limitation to doing so.

### PR DESCRIPTION
**Pull request by LucioFranco**
Closes #1004

Signed-off-by: Lucio Franco <luciofranco14@gmail.com>

### CODE REVIEW

File: .meta/sinks/kafka.toml [hunk 1]
```diff
@@ -13,12 +13,12 @@ write_to_description = "[Apache Kafka][urls.kafka] via the [Kafka protocol][urls
 tls_options = ["+enabled", "-verify"]
 
 [sinks.kafka.options.bootstrap_servers]
-type = "[string]"
+type = "string"
 common = true
-examples = [["10.14.22.123:9092", "10.14.23.332:9092"]]
+examples = ["10.14.22.123:9092, 10.14.23.332:9092"]
```
Comment (lukesteensen) [inline .meta/sinks/kafka.toml]: This example doesn't match the type above (`string` there vs `[string]` here).

Also, most Kafka things don't include a space after the comma in these lists.

**Comment by LucioFranco** [comment]
@lukesteensen yeah, though I think that would differ from what users would expect. I think the best option beyond this change would be to support both.

```toml
servers = "localhost:1,localhost2"

servesr = ["localhost:1", "localhost:2"]
```

File: .meta/sinks/kafka.toml (hunk 1 above)
Comment (LucioFranco) [inline .meta/sinks/kafka.toml]:
```suggestion
examples = ["10.14.22.123:9092,10.14.23.332:9092"]
```

File: .meta/sinks/kafka.toml [hunk 2]
```diff
@@ -13,12 +13,12 @@ write_to_description = "[Apache Kafka][urls.kafka] via the [Kafka protocol][urls
 tls_options = ["+enabled", "-verify"]
 
 [sinks.kafka.options.bootstrap_servers]
-type = "[string]"
+type = "string"
 common = true
-examples = [["10.14.22.123:9092", "10.14.23.332:9092"]]
+examples = ["10.14.22.123:9092,10.14.23.332:9092"]
```
Comment (lukesteensen) [inline .meta/sinks/kafka.toml]: This is still an array :smile:

Comment (LucioFranco) [reply .meta/sinks/kafka.toml]: @lukesteensen I thought its an array to let you show more examples, in the rendered doc it should be a single string?

File: website/docs/reference/sinks/kafka.md [hunk 3]
```diff
@@ -65,7 +65,7 @@ import CodeHeader from '@site/src/components/CodeHeader';
   # REQUIRED - General
   type = "kafka" # must be: "kafka"
   inputs = ["my-source-id"] # example
-  bootstrap_servers = ["10.14.22.123:9092", "10.14.23.332:9092"] # example
+  bootstrap_servers = "10.14.22.123:9092,10.14.23.332:9092" # example
```
Comment (LucioFranco) [inline website/docs/reference/sinks/kafka.md]: @lukesteensen I think this is correct here?

File: .meta/sinks/kafka.toml (hunk 2 above)
Comment (lukesteensen) [reply .meta/sinks/kafka.toml]: Ha. One day I'll understand how these files work...

**Comment by binarylogic** [comment]
> I think the best option beyond this change would be to support both.

Is this the case? If so, this is _not_ a breaking change, right?

**Comment by LucioFranco** [comment]
> Is this the case? If so, this is not a breaking change, right?

Correct, it would be in addition to the current behavior so non breaking.

**Comment by lukesteensen** [comment]
The current diff is a breaking change.

**Comment by LucioFranco** [comment]
Sorry yes, the current diff is breaking, the current one brings both sink/source to use the same config type. In the future, we can add back the list variant without breaking things.

**Pull request merged**